File: cpp/source/event/output/HandoverRejected.cpp
#include "event/output/HandoverRejected.hpp"

using namespace fm;
using namespace fm::event::output;

HandoverRejected::HandoverRejected(const std::string& _message) :
    FacadeEvent(HANDOVER_REJECTED),
    message(_message)
{
}

const std::string& HandoverRejected::getMessage() const
{
    return message;
}

std::string HandoverRejected::toString() const
{
    return "HANDOVER_REJECTED: " + message;
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/output/facade/HandoverRejected.java
package com.fleetmgr.sdk.client.event.output.facade;

/**
 * Created by: Bartosz Nawrot
 * Date: 26.09.2018
 * Description:
 */
public class HandoverRejected extends FacadeEvent {

    private String message;

    public HandoverRejected(String message) {
        super(FacadeEvent.Type.HANDOVER_REJECTED);
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return "HandoverRejected{" +
                "message='" + message + '\'' +
                '}';
    }
}
